src/Auth.ts

```diff
@@ -750,6 +837,63 @@ export class Auth {
       return resource;
     }
   }
+
+  private async getAllConnectionsFromStorage(): Promise<Connection[]> {
+    const connectionStorage = this.getAllConnectionsStorage();
+    const json: string = await connectionStorage.get();
+    return JSON.parse(json);
+  }
+
+  public async switchToConnection(connection: Connection): Promise<void> {
+    this.connection.deactivate();
+    this._connection = Object.assign(this._connection, connection);
+    this._connection.active = true;
+    await this.storeConnectionInfo();
+  }
+
+  public async updateConnection(oldName: string, newName: string): Promise<void> {
+    const allConnections = await this.getAllConnections();
+
+    if (allConnections.filter(c => c.name === newName).length > 0) {
+      throw new CommandError(`The connection name '${newName}' is already in use`);
+    }
+
+    allConnections.forEach(c => {
+      if (c.name === oldName) {
+        c.name = newName;
+      }
+    });
+
+    if (this.connection.name === oldName) {
+      this._connection.name = newName;
+    }
+
+    await this.storeConnectionInfo();
+  }
+
+  public async getConnection(name: string): Promise<Connection> {
+    const allConnections = await this.getAllConnections();
+    const connections = allConnections.filter(i => i.name === name);
+
+    if (!connections || connections.length === 0) {
```

Comment:
If we only care about the first instance, then let's use `allConnections.find` instead of `.filter`. `.find` immediately returns the first matching object or undefined if no such object found.